### PR TITLE
Dataset no attributes method

### PR DESCRIPTION
**Pull request by temanbrcom**
Added new endpoint for pure dataset list using X-IBM-Attributes=dsname in order to be able to list the datasets also if they are archived. Adopted mapper pattern to convert JSON response into internal DTO using MapStruct.

Lightweight dataset list #58

### CODE REVIEW

File: data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetAttributesServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
+/*
```
Comment (stevenhorsman) [inline data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetAttributesServiceTest.java]: Why have you copied ZosmfDataSetServiceTest? where is the not test class for ListDataSetZosmfRequestRunnerTest like ListDataSetAttributesZosmfRequestRunnerTest

File: data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetServiceTest.java [hunk 2]
```diff
@@ -77,10 +74,10 @@ public void testListDataSetMembersRunnerExceptionThrown() throws Exception {
     public void testListDataSetsRunnerValueCorrectlyReturned() throws Exception {
         String filter = "DATA.SET.NAME*";
 
-        DataSetAttributes att1 = DataSetAttributes.builder().name("dataset1").build();
-        DataSetAttributes att2 = DataSetAttributes.builder().name("dataset2").build();
+        DataSet att1 = DataSet.builder().name("dataset1").build();
+        DataSet att2 = DataSet.builder().name("dataset2").build();
 
-        List<DataSetAttributes> expected = Arrays.asList(att1, att2);
+        List<DataSet> expected = Arrays.asList(att1, att2);
```
Comment (stevenhorsman) [inline data-sets-api-server/src/test/java/org/zowe/data/sets/services/zosmf/ZosmfDataSetServiceTest.java]: Please create a new test for list dataset name and exceptions, rather than replacing an existing one

File: data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java [hunk 3]
```diff
@@ -39,13 +39,13 @@ public void testGetValidDataset() throws Exception {
...
-                .recordLength(pdsRequest.getRecordLength()).allocationUnit(pdsRequest.getAllocationUnit())
-                .recordFormat(pdsRequest.getRecordFormat()).allocatedSize(10).used(10).build();
+                    .deviceType("3390").creationDate(today).name(tempDataSet).migrated(false)
+                    .dataSetOrganization(pdsRequest.getDataSetOrganization()).expirationDate("***None***")
+                    .recordLength(pdsRequest.getRecordLength()).allocationUnit(pdsRequest.getAllocationUnit())
+                    .recordFormat(pdsRequest.getRecordFormat()).allocatedSize(10).used(10).build();
 
-            List<DataSetAttributes> actual = getDataSets(tempDataSet).then().statusCode(HttpStatus.SC_OK).extract()
-                .body().jsonPath().getList("", DataSetAttributes.class);
+            List<DataSetAttributes> actual = getDataSetsDetails(tempDataSet).then().statusCode(HttpStatus.SC_OK).extract()
```
Comment (stevenhorsman) [inline data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java]: Please add an integration test for your new endpoint

File: data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java [hunk 4]
```diff
@@ -256,24 +256,24 @@ public void put_data_set_content_with_wrong_type_converted_to_error_message() th
     public void test_get_data_sets_success() throws Exception {
 
         DataSetAttributes cobol = DataSetAttributes.builder().blockSize(133)
-            .dataSetOrganization(DataSetOrganisationType.PO).recordLength(133).recordFormat("FB")
-            .allocationUnit(AllocationUnitType.TRACK).volumeSerial("P4P020").build();
+                .dataSetOrganization(DataSetOrganisationType.PO).recordLength(133).recordFormat("FB")
```
Comment (stevenhorsman) [inline data-sets-api-server/src/test/java/org/zowe/data/sets/controller/DataSetsControllerTest.java]: Create controller unit tests for new endpoint

**Comment by temanbrcom** [comment]
@stevenhorsman Please, verify that all the requirements about tests are met

File: data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/ListDataSetsAttributesZosmfRequestRunner.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/*
```
Comment (stevenhorsman) [inline data-sets-api-server/src/main/java/org/zowe/data/sets/services/zosmf/ListDataSetsAttributesZosmfRequestRunner.java]: Can you refactor ListDataSetAttributesZosmfRequestRunner & ListDataSetZosmfRequestRunner together into a new abstract class given their code is nearly identical, rather than have the copy-pasted code (this can possibly be done for the unit test as well)

File: data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java [hunk 6]
```diff
@@ -32,27 +30,10 @@ public void testGetValidDataset() throws Exception {
...
-            List<DataSetAttributes> actual = getDataSets(tempDataSet).then().statusCode(HttpStatus.SC_OK).extract()
-                .body().jsonPath().getList("items", DataSetAttributes.class);
-
-            // We can't tell the value of some attributes
-            for (DataSetAttributes dataSetAttributes : actual) {
-                dataSetAttributes.setCatalogName(null);
-                dataSetAttributes.setVolumeSerial(null);
-            }
-
+            List<DataSet> actual = getDataSets(tempDataSet).then().statusCode(HttpStatus.SC_OK).extract()
```
Comment (stevenhorsman) [inline data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java]: Please revert this change - this is testing the attributes of the created dataset are correct

File: data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java [hunk 7]
```diff
@@ -32,27 +30,10 @@ public void testGetValidDataset() throws Exception {
         createDataSet(pdsRequest);
 
         try {
+            DataSet expected = DataSet.builder().name(tempDataSet).migrated(false).build();
```
Comment (stevenhorsman) [inline data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java]: Please revert this change and add a new test for your endpoint rather than delete the old one.

File: data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java (hunk 3 above)
Comment (stevenhorsman) [reply data-sets-tests/src/test/java/org/zowe/data/sets/tests/DataSetsGetIntegrationTest.java]: Re-created comment on newer code
